infra/modules/providers/azure/provider/README.md

```diff
@@ -0,0 +1,11 @@
+# Azure Provider
+
+## Summary
```

Comment:
This documentation seems really light. Can you add some more context? For example, which tricky behavior does the null provider assist with? Which limitations are you referring to?

Reply:
Here's an reference of the documentation posted for the lb terraform module https://github.com/Azure/terraform-azurerm-loadbalancer

Reply:
@erikschlegel : 

1. I know its light, but didn't want to mention the service principal stuff as its not part of the PR. The documentation mentions of all the details that are part of the PR. The service principal details should be part of the single-region PR.
2. Null provider, we are not really using now but its mainly to execute scripts as resources that are not defined. I dont think we are using it now so we can go either way, remove it for now and when null resource is needed, add `null provider` in the provider list at that time.
3. To test this locally, I did execute login using service principal, executed `terraform init` and then deployed a resource group using the above provider reference. I can add it part of the PR, but do we actually need that, as we will be creating actual resources and using this provider in the following stories.
```
az login --service-principal -u $APP_ID -p $APP_SECRET --tenant $TENANT_ID
```
```
module "provider" {
  source = "../provider"
}

resource "azurerm_resource_group" "cluster_rg" {
  name     = "cobalt-test-rg"
  location = "eastus"
}
```
I can add this file if you want it to be as part fo the current story.